src/apiserver/api_server_test.cc

```diff
@@ -184,6 +184,81 @@ TEST_F(APIServerTest, query) {
     ASSERT_TRUE(env->cluster_remote->ExecuteDDL(env->db, "drop table demo;", &status));
 }
 
+TEST_F(APIServerTest, parameterizedQuery) {
+    const auto env = APIServerTestEnv::Instance();
+
+    std::string ddl = "create table demo (c1 int, c2 string);";
+    hybridse::sdk::Status status;
+    ASSERT_TRUE(env->cluster_remote->ExecuteDDL(env->db, ddl, &status)) << "fail to create table";
+
+    std::string insert_sql = "insert into demo values (1, \"bb\");";
+    ASSERT_TRUE(env->cluster_sdk->Refresh());
+    ASSERT_TRUE(env->cluster_remote->ExecuteInsert(env->db, insert_sql, &status));
+    insert_sql = "insert into demo values (2, \"bb\");";
+    ASSERT_TRUE(env->cluster_sdk->Refresh());
+    ASSERT_TRUE(env->cluster_remote->ExecuteInsert(env->db, insert_sql, &status));
+
+    {
+        brpc::Controller cntl;
+        cntl.http_request().set_method(brpc::HTTP_METHOD_POST);
+        cntl.http_request().uri() = "http://127.0.0.1:8010/dbs/" + env->db;
+        cntl.request_attachment().append(R"({
+            "sql": "select c1, c2 from demo where c2 = ?;", "mode": "online", "parameters": {
+                "schema": ["STRING"],
```

Comment:
Why need to input schema?

Reply:
because OpenMLDB support `int[16|32|64]` etc., while JSON format only has several types.

Reply:
You can get the exact type by request row, user doesn't have to write the type

Reply:
ref Json2SQLRequestRow

Reply:
we do NOT have a SCHEMA for a query.
e.g. `select c1, c2 from demo where c2 = ?;` needs params of schema `["STRING"]` while `select c1, c2 from demo where c1 = ? AND c2 = ?;` needs `["INT", "STRING"]`

Reply:
We CAN have it, users don't need to do this. see ExecuteProcedure. Plz read it before you stick to your vision

Reply:
https://github.com/4paradigm/OpenMLDB/blob/5ac4667d0e71dc803ddbd04928bc4681b5ba84b6/src/apiserver/api_server_impl.cc#L400-L410

If I understand right, `ExecuteProcedure` gets parameter schema from the procedure already stored in the cluster, so the user doesn't need to input it again.

But for query, we do not have such stored info.

Reply:
You CAN get it in cache too, but you don't try to.
see GetTabletClientForBatchQuery, RouterSQLCache::GetParameterSchema

Reply:
I've tried `SQLRequestRow::GetRequestRow` to extract schema from query, but it turns out `Non-support kFilterPlan Op in online serving`

<details>
<summary>Details</summary>

```
W0912 14:00:26.360630 12917 sql_compiler.cc:329] Fail create sql plan: Non-support kFilterPlan Op in online serving
    (At /workspace/OpenMLDB/hybridse/src/planv2/planner_v2.cc:44)
    (At /workspace/OpenMLDB/hybridse/src/plan/planner.cc:673)
    (At /workspace/OpenMLDB/hybridse/src/plan/planner.cc:510)
    (At /workspace/OpenMLDB/hybridse/src/plan/planner.cc:510)
    (At /workspace/OpenMLDB/hybridse/src/plan/planner.cc:504)
    (Caused by) Non-support kFilterPlan Op in online serving
W0912 14:00:26.361176 12917 sql_cluster_router.cc:1488] fail to explain sql select c1, c2 from demo where c2 = ?;
```

</details>

Reply:
where clause is not supported in online serving mode, try other sql like last join

Reply:
`GetRequestRow` is for request row, not same as parameter row. @vagetablechicken how is parameter row constructed in java sdk ?

Reply:
> `GetRequestRow` is for request row, not same as parameter row. @vagetablechicken how is parameter row constructed in java sdk ?

PreparedStatementImpl, `CreateSQLRequestRowFromColumnTypes` and `ExecuteSQLParameterized` too. What I expect is 
getting parameter schema by sql compiler, like `GetInsertInfo`, not by the request row we sent.
Does it difficult? Or we can do it later. This pr send request row with schema

Reply:
we can keep `schema` field for now. after removing `schema` and the SDK that sends this field will still work.

Reply:
> we can keep `schema` field for now. after removing `schema` and the SDK that sends this field will still work.

@qsliu2017 what do you mean "still work after removing schema"?